app/js/lib/server/settings.js

```diff
@@ -0,0 +1,239 @@
+'use strict';
+
+import EventDispatcher from '../common/event-dispatcher';
+
+// Prefix all entries to avoid collisions.
+const PREFIX = 'cue-';
+
+const ORIGIN = 'https://calendar.knilxof.org';
+
+/**
+ * API version to use (currently not configurable).
+ * @type {number}
+ * @const
+ */
+const API_VERSION = 1;
+
+/**
+ * Regex to match upper case literals.
+ * @type {RegExp}
+ * @const
+ */
+const UPPER_CASE_REGEX = /([A-Z])/g;
+
+const p = Object.freeze({
+  values: Symbol('values'),
+  storage: Symbol('storage'),
+
+  // Private methods.
+  updateSetting: Symbol('updateSetting'),
+  stringToSettingTypedValue: Symbol('stringToSettingTypedValue'),
+  getDefaultSettingValue: Symbol('getDefaultSettingValue'),
+  onStorage: Symbol('onStorage'),
+});
+
+// Definition of all available settings and their default values (if needed).
+const settings = Object.freeze({
+  // String settings.
+  SESSION: Object.freeze({ key: 'session' }),
+  PUSH_ENDPOINT: Object.freeze({ key: 'pushEndpoint' }),
+  PUSH_PUB_KEY: Object.freeze({ key: 'pushPubKey' }),
+  PUSH_AUTH: Object.freeze({ key: 'pushAuth' }),
+});
+
+export default class Settings extends EventDispatcher {
+  constructor(storage = localStorage) {
+    super();
+
+    // Not all browsers have localStorage supported or activated.
+    this[p.storage] = storage || {
+        getItem: () => null,
```

Comment:
grrrr, it looks good to my IDE because the opening braces counts as an indentation level. I agree this looks weird.